### PR TITLE
Added Request Payload to Request Logs

### PR DESCRIPTION
**Pull request by PurpleBriar**
# Type of PR

- [x] Documentation changes
- [x] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## PR Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My code follows the code style of this project.
- [x] I ran lint checks locally prior to submission.
- [x] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Purpose of PR

Added request payload to request logs

## Does this introduce a breaking change

- [ ] YES
- [x] NO

## Validation

- [ ] Unit tests updated and ran successfully
- [x] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes [issue_number (this will automatically close the issue when the PR closes)](https://github.com/retaildevcrews/wcnp/issues/658)

### CODE REVIEW

**Review by jofultz — APPROVED** [review]

File: src/LodeRunner.API/src/Handlers/RequestLogger/RequestLogger.cs [hunk 1]
```diff
@@ -209,10 +213,33 @@ private static string GetClientIp(HttpContext context, out string xff)
...
+        /// Extracts and returns request body
+        /// </summary>
+        /// <param name="request">HttpRequest</param>
+        /// <returns>Request body as object.</returns>
+        private static object LogRequestPayload(HttpRequest request)
+        {
+            if (request.Method == "POST" || request.Method == "PUT")
+            {
+                // Allows using several time the stream in ASP.Net Core
+                request.EnableBuffering();
```
Comment (gortegaMS) [inline src/LodeRunner.API/src/Handlers/RequestLogger/RequestLogger.cs:228]: I think EnableBuffering option should be set at the Startup rather, let's discuss.

File: src/LodeRunner.API/src/Handlers/RequestLogger/RequestLogger.cs [hunk 2]
```diff
@@ -209,10 +213,33 @@ private static string GetClientIp(HttpContext context, out string xff)
...
+
+                StreamReader reader = new (request.Body);
```
Comment (gortegaMS) [inline src/LodeRunner.API/src/Handlers/RequestLogger/RequestLogger.cs:230]: Utilize using() statement for streamReader. e.g 
 using (StreamReader sr = new StreamReader(your source))

**Review by gortegaMS — CHANGES_REQUESTED** [review]

File: src/LodeRunner.API/src/Handlers/RequestLogger/RequestLogger.cs [hunk 3]
```diff
@@ -209,10 +213,33 @@ private static string GetClientIp(HttpContext context, out string xff)
...
             // remove IP6 local address
             return clientIp?.Replace("::ffff:", string.Empty);
+        }
+
+        /// <summary>
+        /// Extracts and returns request body
+        /// </summary>
+        /// <param name="request">HttpRequest</param>
+        /// <returns>Request body as object.</returns>
+        private static object LogRequestPayload(HttpRequest request)
```
Comment (gortegaMS) [inline src/LodeRunner.API/src/Handlers/RequestLogger/RequestLogger.cs:223]: the method does not actually log, I think should be named more like GetPayloadRequestObject or something around that.

**Review by gortegaMS — CHANGES_REQUESTED** [review]

File: src/LodeRunner.API/src/Handlers/RequestLogger/RequestLogger.cs (hunk 3 above)
Comment (PurpleBriar) [reply src/LodeRunner.API/src/Handlers/RequestLogger/RequestLogger.cs:223]: Updated the function name

File: src/LodeRunner.API/src/Handlers/RequestLogger/RequestLogger.cs [hunk 4]
```diff
@@ -209,10 +213,33 @@ private static string GetClientIp(HttpContext context, out string xff)
...
+                // Need to deserialize to remove formatting
+                return JsonSerializer.Deserialize<object>(bodyStr);
+            }
+
+            return null;
         }
 
         // log the request
-        private void LogRequest(HttpContext context, CorrelationVector cv, double ttfb, double duration)
+        private void LogRequest(HttpContext context, CorrelationVector cv, double ttfb, double duration, object payload)
```
[the remark this answers is not in the document]
Comment (PurpleBriar) [reply src/LodeRunner.API/src/Handlers/RequestLogger/RequestLogger.cs:242]: The Invoke function is called on the context object line 148, and the body of the request needs to be extracted before that. I'm not sure why the Invoke function needs to be called before LoqRequest, so I'll see if I can move it around.

[the remark this answers is not in the document]
Comment (gortegaMS) [reply src/LodeRunner.API/src/Handlers/RequestLogger/RequestLogger.cs:242]: I think by calling EnableBuffering() at startup as suggested before can solve this problem

**Pull request merged**